src/test/v1.2/advanced/channels-support-1.2.ts

```diff
@@ -121,23 +125,27 @@ export class ChannelControl1_2 implements ChannelControl<Channel, Context> {
 
   setupAndValidateListener2 = (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
     if (channel) {
-      listener2 = channel.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = channel.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     } else {
-      listener2 = fdc3.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = fdc3.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     }
 
     validateListenerObject(listener2);
   }
 
```

Comment:
I suggest an additional util function here for cases where no context is expected to arrive (which we do in many places and currently output a poor error message)

```suggestion

validateContextIsNotReceived = (channel: Channel, listenerContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
    if (channel) {
      listener1 = channel.addContextListener(listenerContextType, (context) => {
        assert.fail(errorMessage);
        onComplete(context);
      });
    } else {
      listener1 = fdc3.addContextListener(listenerContextType, (context) => {
        assert.fail(errorMessage);
        onComplete(context);
      });
    }
    validateListenerObject(listener1);
  }

```